mssql-plugin/src/main/java/io/cdap/plugin/mssql/SqlServerSourceDBRecord.java

```diff
@@ -67,6 +74,34 @@ protected void handleField(ResultSet resultSet, StructuredRecord.Builder recordB
       }
     }
     switch (sqlType) {
+      case SqlServerSourceSchemaReader.SMALLDATETIME_TYPE:
+        GregorianCalendar gc = new GregorianCalendar(TimeZone.getTimeZone("UTC"));
+        gc.setGregorianChange(new Date(Long.MIN_VALUE));
+        Timestamp timestampSmalldatetime = resultSet.getTimestamp(columnIndex, gc);
+        if (timestampSmalldatetime != null) {
+          ZonedDateTime zonedDateTime = OffsetDateTime.of(timestampSmalldatetime.toLocalDateTime(),
+                                                          OffsetDateTime.now().getOffset())
+                                                          .atZoneSameInstant(ZoneId.of("UTC"));
+          Schema nonNullableSchema = field.getSchema().isNullable() ?
+            field.getSchema().getNonNullable() : field.getSchema();
```

Comment:
you can use fieldSchema here instead of creating another variable nonNullableSchema

Reply:
removed nonNullableSchema